src/main/java/com/academia/library/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.academia.library.service.impl;
+
+import com.academia.library.dto.BookDto;
+import com.academia.library.eception.BookNotFoundException;
+import com.academia.library.mapper.BookMapper;
+import com.academia.library.model.Book;
+import com.academia.library.repository.BookRepository;
+import com.academia.library.service.BookService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
```

Comment:
I recommended you to use readOnly. Please fix it

Reply:
Ok, thank